### DISCUSSION TITLE
Send Later When Email Not Received

### DISCUSSION THREAD

**Discussion by ericp72** [Extended-Thunder/send-later#305] [Q&A]
Has anyone figured out a way to use this Add on to send the email when a certain amount of time has passed with no incoming mail?
Thanks!

**Comment by jph-sendlater** [comment]
Can you explain your use-case in some more detail?

Currently, the closest that exists to what I think you're describing is the "cancel recurrence upon reply" option, which can stop a pending message when it sees a response to a previous message. As one example, you could send a message and then repeat the same message every day until you get a response.

**Reply by jph-sendlater** [reply]
That functionality is not very flexible, though. It can only watch for responses to a previous instance of the same message, not for any arbitrary incoming mail.

**Reply by ericp72** [reply]
Thanks for answering! Yes, what I need to do is have the client send an alert(email) if it does not receive a periodic email from another application. This would alert our engineer that communications might have been lost from the periodic sender.
I was hoping that maybe there was a function that could provide the argument of "send if last timestamp in inbox > 4 hours from current time.
Does that make sense?

**Reply by jph-sendlater** [reply]
That makes sense, but Send Later (and Thunderbird in general) is really not the tool for this job. I would suggest writing a small script that you can run on a schedule. This would be pretty straightforward as a short Python script.

**Comment by jikamens** [comment]
I think what you want here is https://deadmanssnitch.com/, not Send Later.

If you'd rather host it yourself then use a third-party service like Dead Man's Snitch, you should look at [Coal Mine](https://github.com/quantopian/Coal-Mine), which as it happens I wrote and maintain. You can set up Coal Mine on a random server or cloud virtual machine or something and have your periodic task ping a Coal Mine URL each time it runs to notify Coal Mine that it is working. If email is an essential part of this, i.e., you actually want to ensure that the periodic task is able to send out emails not just ping URLs, then you can set up a mail filter on the Coal Mine server that detects the incoming emails and pings the Coal Mine URL when they arrive. Coal Mine will send out notification emails if the periodic task is late, just like Dead Man's Snitch does.

This is not something that we're likely to build into send Later, so I'm going to close this discussion.
